Examples/Java/Sources/Board.java

```diff
@@ -0,0 +1,56 @@
+//
+// Board.java
+// Autogenerated by plank
+//
+// DO NOT EDIT - EDITS WILL BE OVERWRITTEN
+// @generated
+//
+
+package com.pinterest.models;
+
+import android.support.annotation.IntDef;
+import com.google.auto.value.AutoValue;
+import java.net.URI;
+import java.util.Set;
+import android.support.annotation.Nullable;
+import android.support.annotation.StringDef;
+import java.lang.annotation.Retention;
+import java.util.Date;
+import java.util.List;
+import java.lang.annotation.RetentionPolicy;
+import java.util.Map;
+
+@AutoValue
+public abstract class Board {
+
+
+    public abstract @SerializedName("id") @Nullable String identifier();
+    public abstract @SerializedName("contributors") @Nullable Set<User> contributors();
+    public abstract @SerializedName("counts") @Nullable Map<String, Integer> counts();
+    public abstract @SerializedName("created_at") @Nullable Date createdAt();
+    public abstract @SerializedName("creator") @Nullable Map<String, String> creator();
+    public abstract @SerializedName("description") @Nullable String descriptionText();
+    public abstract @SerializedName("image") Image image();
+    public abstract @SerializedName("name") @Nullable String name();
+    public abstract @SerializedName("url") @Nullable URI url();
```

Comment:
It's unsafe to de/serialize platform types like java.net.URI and java.util.Date. When an end user's Android device or JVM updates, the implementation details could change and break your app. Consumers need to register custom adapters for these types.

We should either document this or avoid platform types in favor of strings (maybe with an option to change the type somehow?).

Reply:
That seems like the responsibility of the caller deserializing data to throw-away / discard data from the previous platform version if it is incompatible. 

If it comes down to it we can just specify custom adapter classes and guide users to create them.